ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/user_management_service.proto

```diff
@@ -0,0 +1,175 @@
+
+    // List the set of all rights granted to a user.
+    rpc ListUserRights (ListUserRightsRequest) returns (ListUserRightsResponse);
+}
+
+
+// Users and rights
+///////////////////
+
+// Users are are used to manage the rights given to authenticated Ledger API clients.
+// They are stored and managed per participant node.
+//
+// TLS client certificates authenticate the user specified by the certificates common name.
+// JWT tokens authenticate the user specified by the JWT tokens 'sub' (subject) claim.
+message User {
+    // The user identifier, which must be a non-empty string of at most 64
+    // characters matching the regexp [a-z0-9-_.], i.e. lower-case characters, digits,
+    // dashes or dots. Double-dots, double-dashes, and double-underscores are not allowed.
+    string id = 1;
+
+    // The primary party as which this user reads and acts by default on the ledger
+    // _provided_ it has the corresponding ``CanReadAs(primary_party)`` or 
+    // ``CanActAs(primary_party)`` rights.
+    string primary_party = 2;
+}
+
+// A right granted to a user.
+message Right {
+    // The user is allowed to administrate the participant node.
+    message ParticipantAdmin {}
+
+    // The user can authorize commands for the given party.
+    message CanActAs {
+        string party = 1;
+    }
+
+    // The user can read ledger data visible to the given party.
+    message CanReadAs {
+        string party = 1;
+    }
+
+    oneof kind {
+        ParticipantAdmin participant_admin = 1;
+        CanActAs can_act_as = 2;
+        CanReadAs can_read_as = 3;
+    }
+}
+
+
+// RPC requests and responses
+/////////////////////////////
+
+// Required authorization: ``HasRight(ParticipantAdmin)``
+message CreateUserRequest {
+    // The user to create.
+    User user = 1;
+
+    // The rights to be assigned to the user upon creation,
+    // which SHOULD include appropriate rights for the ``user.primary_party``.
+    repeated Right rights = 2;
+}
+
+// Required authorization: ``HasRight(ParticipantAdmin) OR IsAuthenticatedUser(user_id)``
+message GetUserRequest {
+    // The user whose data we want to retrieve.
+    // If set to empty string (the default), then the data for the authenticated user will be retrieved.
+    string user_id = 1;
+}
+
+// Required authorization: ``HasRight(ParticipantAdmin)``
+message DeleteUserRequest {
+    string user_id = 1;
+}
+
+// Required authorization: ``HasRight(ParticipantAdmin)``
+message ListUsersRequest {
+    // TODO: add pagination, cf. https://cloud.google.com/apis/design/design_patterns#list_pagination
+}
+
+message ListUsersResponse {
+    repeated User users = 1;
+}
+
+// Add the rights to the set of rights granted to the user.
+//
+// Required authorization: ``HasRight(ParticipantAdmin)``
+message GrantUserRightsRequest {
+    string user_id = 1;
+
+    repeated Right rights = 2;
+}
+
+message GrantUserRightsResponse {
```

Comment:
Maybe it's useful to also include all rights (or just the set of already existing rights) in the response. If I think about the usage of this in a GUI, I'd probably want to refresh the user's rights with a definitive authoritative answer about all rights the user has.

Reply:
The reason why these rights are not included is that this won't scale to large sets of rights.